File: src/KubeOps.Templates/Templates/Operator.CSharp/Webhooks/DemoValidator.cs
using KubeOps.Operator.Webhooks;
using GeneratedOperatorProject.Entities;

namespace GeneratedOperatorProject.Webhooks;

public class DemoValidator : IValidationWebhook<V1DemoEntity>
{
    public AdmissionOperations Operations => AdmissionOperations.Create;

    public ValidationResult Create(V1DemoEntity newEntity, bool dryRun)
        => newEntity.Spec.Username == "forbiddenUsername"
            ? ValidationResult.Fail(StatusCodes.Status400BadRequest, "Username is forbidden")
            : ValidationResult.Success();
}

File: _old/src/KubeOps.Templates/Templates/Operator.FSharp/Webhooks/DemoValidator.fs
namespace GeneratedOperatorProject.Webhooks

open GeneratedOperatorProject.Entities
open KubeOps.Operator.Webhooks
open Microsoft.AspNetCore.Http

type DemoValidator() =
    interface IValidationWebhook<V1DemoEntity> with
        member this.Operations = AdmissionOperations.Create

        member this.Create(newEntity, _) =
            if newEntity.Spec.Username = "forbiddenUsername"
            then ValidationResult.Fail(StatusCodes.Status400BadRequest, "Username is forbidden")
            else ValidationResult.Success()
